### PR TITLE
move gatk4/mergevcfs to nf-test

### PR DESCRIPTION
**Pull request by matthdsm**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/gatk4/mergevcfs/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                input[0] = [ [ id:'test' ], [ file(params.test_data['homo_sapiens']['genome']['dbsnp_146_hg38_vcf_gz'], checkIfExists: true), file(params.test_data['homo_sapiens']['genome']['gnomad_r2_1_1_vcf_gz'], checkIfExists: true) ]]
+                input[1] = [ [],[]]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+            )
```
Comment (nvnieuwk) [inline modules/nf-core/gatk4/mergevcfs/tests/main.nf.test:69]: Can you also check that the outputs get created in a stub run?

File: modules/nf-core/gatk4/mergevcfs/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                input[0] = [ [ id:'test' ], [ file(params.test_data['homo_sapiens']['genome']['dbsnp_146_hg38_vcf_gz'], checkIfExists: true), file(params.test_data['homo_sapiens']['genome']['gnomad_r2_1_1_vcf_gz'], checkIfExists: true) ]]
+                input[1] = [ [],[]]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(file(process.out.vcf.get(0).get(1)).list()).match() }
```
Comment (nvnieuwk) [inline modules/nf-core/gatk4/mergevcfs/tests/main.nf.test:46]:
```suggestion
                { assert snapshot(file(process.out.vcf.get(0).get(1)).list()).match() },
```

File: modules/nf-core/gatk4/mergevcfs/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                input[0] = [ [ id:'test' ], [ file(params.test_data['homo_sapiens']['genome']['dbsnp_146_hg38_vcf_gz'], checkIfExists: true), file(params.test_data['homo_sapiens']['genome']['gnomad_r2_1_1_vcf_gz'], checkIfExists: true) ]]
+                input[1] = [ [], file(params.test_data['homo_sapiens']['genome']['genome_dict'], checkIfExists: true)]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(file(process.out.vcf.get(0).get(1)).list()).match() }
```
Comment (nvnieuwk) [inline modules/nf-core/gatk4/mergevcfs/tests/main.nf.test:25]:
```suggestion
                { assert snapshot(file(process.out.vcf.get(0).get(1)).list()).match() },
```

File: modules/nf-core/gatk4/mergevcfs/tests/nextflow.config
Comment (maxulysse) [inline modules/nf-core/gatk4/mergevcfs/tests/nextflow.config:1]: is this file useful?

Comment (matthdsm) [reply modules/nf-core/gatk4/mergevcfs/tests/nextflow.config:1]: not atm, but it's autogenerated by #tools and it was there for the pytest

Comment (maxulysse) [reply modules/nf-core/gatk4/mergevcfs/tests/nextflow.config:1]: then let's delete it

Comment (matthdsm) [reply modules/nf-core/gatk4/mergevcfs/tests/nextflow.config:1]: done!

**Comment by matthdsm** [comment]
Can I get an approval here?

**Review by maxulysse — APPROVED** [review]

**Comment by matthdsm** [comment]
Thanks @maxulysse !

**Pull request merged**